### PR TITLE
Update codec

### PR DESCRIPTION
**Pull request by Nujanauss**
Make sure commands work properly (have not checked all)

Also fixed a couple of logging bugs, and added a MainActivity that we can use to debug (which is certainly not fully functional right now)

### CODE REVIEW

File: mentalabMobileApi/src/main/java/com/mentalab/io/BluetoothManager.java [hunk 1]
```diff
@@ -32,7 +32,9 @@ public static Set<BluetoothDevice> getBondedDevices() throws NoBluetoothExceptio
   public static BluetoothAdapter getBluetoothAdapter() throws NoBluetoothException {
     final BluetoothAdapter btAdapter = BluetoothAdapter.getDefaultAdapter();
     if (btAdapter == null) {
-      throw new NoBluetoothException("Bluetooth service not available", null);
+      throw new NoBluetoothException("Bluetooth service not available. Exiting.");
+    }  else if (!btAdapter.isEnabled()) {
+      throw new NoBluetoothException("Bluetooth is not enabled. Exiting.");
```
Comment (salman2135) [inline mentalabMobileApi/src/main/java/com/mentalab/io/BluetoothManager.java:37]: what is the difference between service not available and service not enabled?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -6,6 +6,26 @@
   <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
                    android:maxSdkVersion="28"/>
 
+  <!-- Request legacy Bluetooth permissions on older devices. -->
+  <uses-permission android:name="android.permission.BLUETOOTH"
```
Comment (salman2135) [inline app/src/main/AndroidManifest.xml:10]: do we need API level 30 here?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -6,6 +6,26 @@
...
+  <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"
+                   android:maxSdkVersion="30" />
+
+  <!-- Include "neverForLocation" only if you can strongly assert that
+          your app never derives physical location from Bluetooth scan results. -->
+  <uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
+
+  <!-- Needed only if your app makes the device discoverable to Bluetooth
+       devices. -->
+  <uses-permission android:name="android.permission.BLUETOOTH_ADVERTISE" />
```
Comment (salman2135) [inline app/src/main/AndroidManifest.xml:21]: do we need to advertise to other devices?

File: mentalabMobileApi/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -3,4 +3,5 @@
     package="com.mentalab">
 
     <uses-permission android:name="android.permission.BLUETOOTH" />
+    <uses-permission android:name="android.permission.BLUETOOTH_CONNECT"/>
```
Comment (salman2135) [inline mentalabMobileApi/src/main/AndroidManifest.xml:6]: Not sure if we need connect permission here..is this related to API level 30?

File: mentalabMobileApi/src/main/java/com/mentalab/MentalabCommands.java [hunk 5]
```diff
@@ -278,11 +292,11 @@ private static Map<Topic, Uri> generateFiles(RecordTask recordSubscriber) throws
     return androidFileGenerator.generateFiles(directory, filename);
   }
 
-  public void clearDeviceList() {
+  public static void clearDeviceList() {
```
Comment (salman2135) [inline mentalabMobileApi/src/main/java/com/mentalab/MentalabCommands.java:295]: Is clear devices list necessary?

File: mentalabMobileApi/src/main/java/com/mentalab/io/BluetoothManager.java (hunk 1 above)
Comment (Nujanauss) [reply mentalabMobileApi/src/main/java/com/mentalab/io/BluetoothManager.java:37]: service not available -> it's not possible to turn on bluetooth
service not enabled -> it's possible to turn on bluetooth

File: mentalabMobileApi/src/main/java/com/mentalab/MentalabCommands.java (hunk 5 above)
Comment (Nujanauss) [reply mentalabMobileApi/src/main/java/com/mentalab/MentalabCommands.java:295]: I think we may need it in the future. But I could be wrong. Let's leave it for now, but good point

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (Nujanauss) [reply app/src/main/AndroidManifest.xml:21]: Good catch - misread

File: mentalabMobileApi/src/main/AndroidManifest.xml (hunk 4 above)
Comment (salman2135) [reply mentalabMobileApi/src/main/AndroidManifest.xml:6]: I believe it should be fine with old permissions.  Other permissions are redundant(on Android API level 30 or 28 as target).

**Review by salman2135 — APPROVED** [review]
